pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,232 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument, RangeException
+from pymeasure.adapters import PrologixAdapter
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+
+import numpy as np
+import time
+from io import BytesIO
+import re
+
+
+class Sorensen30035E(Instrument):
+    """ Represents the Sorensen300-3.5E power supply with GPIB option and provides a
+    high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        sorensen = Sorensen30035E("GPIB::1")
+
+        sorensen.enable_source()                # Enables output of current
+        sorensen.ramp_to_current(200,200)       # Ramp to 200 V in 200 seconds nominally
+        
+        print(sorensen.current)                 # Prints the actual current output in Amps
+
+        sorensen.shutdown()                     # Ramps the current to 0 mA and disables output
+
+    """
+
+    source_enable = Instrument.control(
+        "OUTPut:PROTection:STATe?", "OUTPut:PROTection:STATe %d",
+        """A boolean property that sets the ouput to zero or the programmed value,
+        openning or closing the isolation relay. *RST value is ON. CAUTION: Ensure
+        that suitable delays are incorporated to preclude hot switching of the isolation relay.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True
+    )
+
+    source_isolation = Instrument.control(
+        "OUTPut:ISOLation?", "OUTPut:ISOLation %d",
+        """A boolean property that controls whether the isolation relay is enabled. 
+        DO NOT disable if the current is non-zero, i.e. no hot-switching""",
```

Comment:
This seems a (potentially damaging) hardware-level operation constraint. Maybe it would be better to implement `source_isolation` as a traditional property, then you can add some safeguarding logic in the setter, e.g. if the user wants to disable, check first if the instrument's current is below a certain threshold, if not raise an exception or warn otherwise.